pkg/store/cache/cache.go

```diff
@@ -0,0 +1,325 @@
+package storecache
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	lru "github.com/hashicorp/golang-lru/simplelru"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/tsdb/labels"
+)
+
+const (
+	cacheTypePostings string = "Postings"
+	cacheTypeSeries   string = "Series"
+
+	sliceHeaderSize = 16
+)
+
+type cacheItem struct {
+	block ulid.ULID
+	key   interface{}
+}
+
+func (c cacheItem) keyType() string {
+	switch c.key.(type) {
+	case cacheKeyPostings:
+		return cacheTypePostings
+	case cacheKeySeries:
+		return cacheTypeSeries
+	}
+	return "<unknown>"
+}
+
+func (c cacheItem) size() uint64 {
+	switch k := c.key.(type) {
+	case cacheKeyPostings:
+		// ULID + 2 slice headers + number of chars in value and name.
+		return sliceHeaderSize + 2*16 + uint64(len(k.Value)+len(k.Name))
```

Comment:
Comment and code don't quite match up ("ulid + 2 slice headers" vs `sliceHeaderSize + 2*16`).